pkg/apis/agones/v1/gameserver.go

```diff
@@ -596,6 +597,11 @@ func (gs *GameServer) Pod(sidecars ...corev1.Container) (*corev1.Pod, error) {
 		Spec:       *gs.Spec.Template.Spec.DeepCopy(),
 	}
 
+	if len(pod.Spec.Hostname) == 0 {
+		// replace . with - since it must match RFC 1123
+		pod.Spec.Hostname = strings.ReplaceAll(gs.ObjectMeta.Name, ".", "-")
+	}
+
 	gs.podObjectMeta(pod)
```

Comment:
Just adding this as a formal review comment. I think since this is altering the generated pod in a possibly significant way, we should add a feature gate. It can go straight to `beta`, IMO - but especially in a world where we're releasing straight from `main`, I think we should be a little cautious.

Reply:
Sounds good to me - I'll make the change.

Reply:
...and done!